roles/mdc/tasks/generate-manifest.yml

```diff
@@ -0,0 +1,118 @@
+  register: mss_route_cmd
+
+- set_fact:
+    mss_route: "https://{{mss_route_cmd.stdout}}"
+
+- name: Set Mobile Security Service component
+  set_fact:
+    mss_manifest:
+      - name: Mobile Security Service
+        version: "{{ mss_version }}"
+        host: "{{ mss_route }}"
+        mobile: true
+        type: "security"
+
+- set_fact:
+    mobile_components: "{{mobile_components}} + {{ mss_manifest }}"
+
+#unified push server(ups)
+- name: Unified Push Server
+  block:
+    - name: Get Unified Push Server route
+      shell: oc get route/unifiedpush-unifiedpush-proxy  -o template --template \{\{.spec.host\}\} -n {{ mobile_services_namespace }}
+      register: ups_route_cmd
+    - set_fact:
+        ups_route: "https://{{ups_route_cmd.stdout}}"
+    - name: Set Unified Push Server component
+      set_fact:
+        ups_manifest:
+          - name: Unified Push Server
+            version: "{{ ups }}"
+            host: "{{ ups_route }}"
+            mobile: true
+            type: "push"
+    - set_fact:
+        mobile_components: "{{ mobile_components }} + {{ ups_manifest }}"
+
+#IDM
+- name: Identity Management Service
+  block:
+    - name: Get IDM server route
+      shell: oc get route/sso  -o template --template \{\{.spec.host\}\} -n {{ mobile_services_namespace }}
+      register: sso_route_cmd
+    - set_fact:
+        sso_route: "https://{{sso_route_cmd.stdout}}"
+    - name: Set IDM component
+      set_fact:
+        idm_manifest:
+          - name: Identity Management
+            version: "{{ idm }}"
+            host: "{{ sso_route }}"
+            mobile: true
+            type: "keycloak"
+    - set_fact:
+        mobile_components: "{{ mobile_components }} + {{ idm_manifest }}"
+
+# mobile developer console (mdc)
+- name: Mobile Developer Console
+  block:
+    - name: Get mdc secure route
+      shell: "oc get route/mdc-mdc-proxy -o jsonpath='{.spec.host}' -n {{ mobile_services_namespace }}"
+      register: mdc_route_cmd
+    - set_fact:
+        mdc_route: "https://{{mdc_route_cmd.stdout}}"
+    - name: Set Mobile Developer Console component
+      set_fact:
+        mdc_manifest:
+          - name: Mobile Developer Console
+            version: "{{ mdc }}"
+            host: "{{ mdc_route }}"
+    - set_fact:
+        mobile_components: "{{ mobile_components }} + {{ mdc_manifest }}"
+
+- name: generate manifest file
+  template: 
+    src: manifest.j2
+    dest: /tmp/manifest.json
+
+- name: Create manifest secret file
+  shell: oc create secret generic manifest --from-file=generated_manifest=/tmp/manifest.json --type=application/json -n {{ mobile_services_namespace }}
+  register: result
+  failed_when: result.stderr != '' and 'AlreadyExists' not in result.stderr         
+
+- name: Get mdc deployment config
+  shell: "oc get dc/mdc -o json -n {{ mobile_services_namespace }}"
+  register: mdc_get_dc_cmd
+
+- name: patch mdc dc
```

Comment:
I could be missing something but is it necessary to do this with the installer or can it be done in the environment variables in the deployment via the operator?

Reply:
for now, yes, we need to do it through the installer. But you are right, ideally this can be done through the MDC operator: we can specify the name of the manifest in the CR and the operator should be able to configure MDC to mount it. I will create a JIRA for it.

Reply:
jira created: https://issues.jboss.org/browse/AEROGEAR-10019